File: c++/ZmqStormExample.cpp
/*

*/

#include "ZmqStormClient.h"

int main (int argc, char *argv [])
{
    ZmqStormClient *stormClient;

    if (argc != 3) {
        printf ("usage: ZmqStormExample <ip> <port>\n");
        return 1;
    }

    stormClient = initZmq(argv[1], argv[2]);
    if (!stormClient) {
        printf ("error in zmq_socket: %s\n", zmq_strerror (errno));
        return 1;
    }

    while(true)
    {
        char * msg = receivemsg(stormClient);
        printf("%s", msg);
    }
    return 0;
}

File: java/ZmqStormExample.java
/*
  消息队列例子，输入参数为服务器的IP地质、端口号。
*/

  import org.zeromq.ZMQ;

  class ZmqStormExample
  {
    public static void main (String [] args)
    {
      if (args.length != 2) {
        System.out.println ("usage: ZmqStormExample <DC-Storm> <serverip> " +
          "<serverport>");
        return;
      }

      ZmqStormClient client = new ZmqStormClient(args[0], args[1]);

      while(true)
      {
          byte[] msg= client.receive(); // 阻塞获取新消息
          System.out.println (new String(msg));
      }
    }
  }
